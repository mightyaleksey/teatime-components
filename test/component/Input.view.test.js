'use strict';

const Input = require('../../component/Input');
const React = require('react');
const renderer = require('react-test-renderer');

test('defaultProp #size should have a value `m`', () => {
  expect(Input.defaultProps.size).toBe('m');
});

test('Renders correctly with default props', () => {
  const tree = renderer.create(
    <Input
      name='control'/>
  );

  expect(tree).toMatchSnapshot();
});

test('Renders correctly with defined props', () => {
  const tree = renderer.create(
    <Input
      autoComplete='off'
      autoFocus
      className='outer'
      disabled
      id='controls-id'
      maxLength='255'
      name='control'
      placeholder='__'
      readOnly
      type='password'
      value='initial'/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
