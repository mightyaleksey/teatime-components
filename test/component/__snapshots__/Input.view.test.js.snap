// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with default props 1`] = `
<span
  className="wrapper"
  onChange={undefined}
>
  <input
    autoComplete={undefined}
    autoFocus={undefined}
    className="control"
    disabled={undefined}
    id={undefined}
    maxLength={undefined}
    name="control"
    onBlur={undefined}
    onChange={[Function]}
    onFocus={undefined}
    placeholder={undefined}
    readOnly={undefined}
    type="text"
    value=""
  />
</span>
`;

exports[`Renders correctly with defined props 1`] = `
<span
  className="wrapper outer"
  onChange={undefined}
>
  <input
    autoComplete="off"
    autoFocus={true}
    className="control"
    disabled={true}
    id="controls-id"
    maxLength="255"
    name="control"
    onBlur={undefined}
    onChange={[Function]}
    onFocus={undefined}
    placeholder="__"
    readOnly={true}
    type="password"
    value="initial"
  />
</span>
`;
